openshift-client/src/main/java/io/fabric8/openshift/client/DefaultOpenShiftClient.java

```diff
@@ -640,7 +639,9 @@ public ExtensionsAPIGroupClient extensions() {
 
   @Override
   public VersionInfo getVersion() {
-    return new OpenShiftClusterOperationsImpl(httpClient, getConfiguration(), OpenShiftClusterOperationsImpl.OPENSHIFT_VERSION_ENDPOINT).fetchVersion();
+    final VersionInfo versionInfo = new OpenShiftClusterOperationsImpl(httpClient,
+      getConfiguration(), OpenShiftClusterOperationsImpl.OPENSHIFT_VERSION_ENDPOINT).fetchVersion();
+    return versionInfo == null ? DefaultKubernetesClient.getVersion(this) : versionInfo;
```

Comment:
Maybe it's not necessary to create the static method.
Would this work too?
```suggestion
    return versionInfo == null ? super.getVersion() : versionInfo;
```

Reply:
I realized this while driving to pick up my kids from school 😅